denizen_scripts/survival/repo-link/items/system/soul_item_system.dsc

```diff
@@ -247,58 +337,78 @@ item_with_soul:
-    - define rarity <[soul_item].nbt[rarity]>
-    - define item <list[<[item1]>|<[item2]>].filter[has_nbt[soul].not].get[1].material.name>
-    - define enchants <list[<[item1]>|<[item2]>].filter[has_nbt[soul].not].get[1].enchantments.with_levels.escaped||<list[]>>
-    - determine <proc[item_create_soul_item].context[<[item]>|<[soul_type]>|<[soul_level]>|<[rarity]>|<[enchants]>]>
+
+    - define soul_item <list[<[item1]>].include[<[item2]>].filter[has_nbt[soul]].first>
+    - define contexts <list.include[<list[<[item1]>|<[item2]>].filter[has_nbt[soul].not].first.material.name>]>
+    - define contexts <[contexts].include[<[soul_item].nbt[soul].get[type]>]>
+    - define contexts <[contexts].include[<[soul_item].nbt[soul].get[level]>]>
+    - define contexts <[contexts].include[<[soul_item].nbt[rarity]>]>
+    - define contexts <[context].include[<list_single[<list[<[item1]>|<[item2]>].parse_tag[<[parse_value].enchantments.with_levels>].combined.deduplicate>]>]>
+    - determine <proc[item_create_soul_item].context[<[Contexts]>]>
 
 item_create_soul_item:
   type: procedure
-  definitions: material|soul_type|level|rarity|enchants_escaped
+  definitions: material|soul_type|level|rarity|enchantments
   script:
-    - if <[material].matches[<script[item_system_global_data].yaml_key[regex_type_check.weapon]>]>:
-      - define base <script[item_system_global_data].yaml_key[defaults.damage.<[material]>]>
-    - foreach <script[item_system_global_data].yaml_key[soul_names.<[rarity]>.<[soul_type]>]> as:modifier:
-      - define stat <[modifier].before[/]>
-      - define type <[modifier].after[/]>
+    - if <[material].matches[<script[item_system_global_data].data_key[regex_type_check.weapon]>]>:
+      - define base <script[item_system_global_data].data_key[defaults.damage.<[material]>]>
+    - foreach <script[item_system_global_data].data_key[soul_names.<[rarity]>.<[soul_type]>]> key:stat as:type:
       - if <[type]> == buff:
-        - define buffs:|:<[stat]>,<script[item_system_global_data].yaml_key[calculations.<[stat]>].parsed>
+        - define buffs:->:<map.with[<[stat]>].as[<script[item_system_global_data].parsed_key[calculations.<[stat]>]>]>
       - else:
-        - define debuffs:|:<[stat]>,<script[item_system_global_data].yaml_key[calculations.<[stat]>].parsed>
-    - if <[enchants_escaped].unescaped.as_list.size||0> >= 1:
-      - define item_to_build <item[<[material]>].with[enchantments=<[enchants_escaped].unescaped>;nbt=buffs/<[buffs].escaped||none>|debuffs/<[debuffs].escaped||none>|rarity/<[rarity]>|soul_level/<[level]>|active_soul/<[soul_type]>]>
+        - define debuffs:->:<map.with[<[stat]>].as[<script[item_system_global_data].parsed_key[calculations.<[stat]>]>]>
+    - if !<[enchantments].is_empty>:
+      - define item_to_build <item[<[material]>].with[enchantments=<[enchantments]>;nbt=buffs/<[buffs]||none>|debuffs/<[debuffs]||none>|rarity/<[rarity]>|soul_level/<[level]>|active_soul/<[soul_type]>]>
```

Comment:
This would never return invalid; Items without enchantments return an empty list for this sub-tag. Removed check for invalid / more than 0 enchantment size for whether the list is empty, or not.
Also removed escaping as this is preserved now within lists.